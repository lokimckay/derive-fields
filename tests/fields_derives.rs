mod common;

use common::*;
use derive_fields::Fields;

#[derive(Fields)]
#[fields_derives(Debug)]
pub struct ExampleStruct {
    _name: String,
    _category: Category,
    _really_really_long_key: bool,
}

fn main() {
    // Confirm that field no longer implements clone
    let field = ExampleStructField::Name("Albert".to_string());
    let _ = field.clone();
}
