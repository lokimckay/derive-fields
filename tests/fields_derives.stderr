error[E0599]: no method named `clone` found for enum `ExampleStructField` in the current scope
  --> tests/fields_derives.rs:17:19
   |
 6 | #[derive(Fields)]
   |          ------ method `clone` not found for this enum
...
17 |     let _ = field.clone();
   |                   ^^^^^ method not found in `ExampleStructField`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `clone`, perhaps you need to implement it:
           candidate #1: `Clone`
